src/operations/Operation.cpp

```diff
@@ -249,4 +249,29 @@ void Operation::addDepthContribution(std::vector<std::size_t>& depths) const {
   }
 }
 
+void Operation::addTwoQubitDepthContribution(
+    std::vector<std::size_t>& depths) const {
+  if (type == Barrier) {
+    return;
+  }
+
+  std::size_t maxDepth = 0;
+  for (const auto& target : getTargets()) {
+    maxDepth = std::max(maxDepth, depths[target]);
+  }
+  for (const auto& control : getControls()) {
+    maxDepth = std::max(maxDepth, depths[control.qubit]);
+  }
+
+  if (!getControls().empty()) {
```

Comment:
There are two-qubit gates that do not have a control (e.g. SWAP gate) but are still two-qubit gates.